docs/guide/development/external-library-delegation.md

```
@@ -0,0 +1,346 @@
+### Two Kinds of Steps
+
+There are two kinds of steps that can be defined in YAML:
+
+1. **Internal Steps**: Refer to services within the application annotated with @PipelineStep
+2. **Delegated Steps**: Refer to operators that are NOT annotated with @PipelineStep
+
+### Type Layers for Delegated Steps
+
+When delegation is used, there are four conceptual layers:
+
+```
```

markdownlint-cli2 (0.20.0)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)